cf-java-logging-support-servlet/src/main/java/com/sap/hcp/cf/logging/servlet/filter/RequestLoggingFilterBase.java

```diff
@@ -0,0 +1,172 @@
+package com.sap.hcp.cf.logging.servlet.filter;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.MDC;
+
+import com.sap.hcp.cf.logging.common.LogContext;
+import com.sap.hcp.cf.logging.common.LogOptionalFieldsSettings;
+import com.sap.hcp.cf.logging.common.request.HttpHeader;
+import com.sap.hcp.cf.logging.common.request.HttpHeaders;
+import com.sap.hcp.cf.logging.common.request.RequestRecord;
+
+/**
+ * <p>
+ * A simple servlet filter that logs HTTP request processing info. It will read
+ * several HTTP Headers and store them in the SLF4J MDC, so that all log
+ * messages created during request handling will have those additional fields.
+ * It will also instrument the request to generate a request log containing
+ * metrics such as request and response sizes and response time. This
+ * instrumentation can be disabled by denying logs from {@link RequestLogger}
+ * with marker "request".
+ * </p>
+ * <p>
+ * This filter will generate a correlation id, from the HTTP header
+ * "X-CorrelationID" falling back to "x-vcap-request-id" if not found or using a
+ * random UUID. The correlation id will be added as an HTTP header
+ * "X-CorrelationID" to the response if possible.
+ * </p>
+ * <p>
+ * To use the filter, it needs to be added to the servlet configuration. It has
+ * a default constructor to support web.xml configuration. There are several
+ * other constructors that support some customization, in case dynamic
+ * configuration (e.g. Spring Boot) is used. You can add further customization
+ * by subclassing the filter and overwrite its methods.
+ * </p>
+ */
+public class RequestLoggingFilterBase implements Filter {
```

Comment:
I am not quite sure about the naming. I would prefer to have this end on Filter, since this is the implemented interface. What do you think about `BaseRequestLoggingFilter`?

Reply:
done with latest change

Reply:
That is also fine with me.